std/os/windows.zig

```diff
@@ -765,7 +765,7 @@ inline fn MAKELANGID(p: USHORT, s: USHORT) LANGID {
 pub fn unexpectedError(err: DWORD) std.os.UnexpectedError {
     if (std.os.unexpected_error_tracing) {
         var buf: LPSTR = undefined;
-        const len = kernel32.FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, null, err, MAKELANGID(LANG.LANG_NEUTRAL, LANG.SUBLANG_DEFAULT), buf, 0, null);
+        const len = kernel32.FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, null, err, MAKELANGID(LANG.LANG_NEUTRAL, LANG.SUBLANG_DEFAULT), @ptrCast(LPSTR, &buf), 0, null);
```

Comment:
I think we should use `FormatMessageW` so that this does not introduce a dependency on system configuration. See #534.

Reply:
i was going to use `FormatMessageA` but i wasn't sure how that would work when outputting the message. would we use `std.unicode` to convert it?

Reply:
i ended up switching to using fixed-length buffers with lengths based on the error messages defined in `std/os/windows/error.zig`.